### PR TITLE
Scope variables in move.c

### PR DESCRIPTION
**Pull request by alkino**

### CODE REVIEW

File: src/nvim/move.c [hunk 1]
```diff
@@ -74,8 +71,9 @@ static void comp_botline(win_T *wp)
   }
 
   for (; lnum <= wp->w_buffer->b_ml.ml_line_count; ++lnum) {
-    last = lnum;
-    folded = FALSE;
+    int n;
+    linenr_T last = lnum;
+    int folded = FALSE;
```
Comment (ZyX-I) [inline src/nvim/move.c]: You can’t add line with FALSE without failing the clint build. Or touch any line near line with FALSE as well.

Comment (alkino) [inline src/nvim/move.c]: OK. I let this in suspens. I will do a patch that remove all FALSE / TRUE of this file.

**Comment by alkino** [comment]
More things to pass linter.

File: src/nvim/move.c [hunk 2]
```diff
@@ -74,15 +71,15 @@ static void comp_botline(win_T *wp)
   }
 
   for (; lnum <= wp->w_buffer->b_ml.ml_line_count; ++lnum) {
-    last = lnum;
-    folded = FALSE;
-    if (hasFoldingWin(wp, lnum, NULL, &last, TRUE, NULL)) {
+    int n;
+    linenr_T last = lnum;
+    bool folded = hasFoldingWin(wp, lnum, NULL, &last, true, NULL);
+    if (folded)
```
Comment (bfredl) [inline src/nvim/move.c]: needs braces

File: src/nvim/move.c [hunk 3]
```diff
@@ -1345,20 +1309,15 @@ void scroll_cursor_top(int min_scroll, int always)
   new_topline = top + 1;
 
   /* count filler lines of the cursor window as context */
-  i = diff_check_fill(curwin, curwin->w_cursor.lnum);
-  used += i;
-  extra += i;
+  int extra = diff_check_fill(curwin, curwin->w_cursor.lnum);
+  used += extra;
```
Comment (justinmk) [inline src/nvim/move.c]: Can you explain why it was ok to remove `i`?

File: src/nvim/move.c [hunk 4]
```diff
@@ -1345,20 +1309,15 @@ void scroll_cursor_top(int min_scroll, int always)
...
 
   /*
    * Check if the lines from "top" to "bot" fit in the window.  If they do,
    * set new_topline and advance "top" and "bot" to include more lines.
    */
   while (top > 0) {
-    if (hasFolding(top, &top, NULL))
-      /* count one logical line for a sequence of folded lines */
```
Comment (justinmk) [inline src/nvim/move.c]: why remove the comment?

**Comment by justinmk** [comment]
@alkino Good choices here, thanks. In the future it would really help to add a couple tests, or else verify that the affected functions have full coverage at https://coveralls.io/github/neovim/neovim

File: src/nvim/move.c (hunk 3 above)
Comment (alkino) [inline src/nvim/move.c]: So.

i is here only for a temporary var that is equal to extra. Why? Look, extra is initialized during declaration to 0.
extra += i => extra = extra + i => extra = 0 + i => extra = i
If you look some lines below, you can see that i is set at the beginning of while. So the value of i here is not used anymore. 

I find it more readable (imho)
"extra" is "diff_check_fill" and we add it to "used"
than "i" is "diff_check_fill" and add it to "extra" and "used"

File: src/nvim/move.c (hunk 4 above)
Comment (alkino) [inline src/nvim/move.c]: Ok I will reput it.

File: src/nvim/move.c (hunk 3 above)
Comment (dnmario) [inline src/nvim/move.c]: Dear Nicolas Cornu,

The email sent to danielm@tinyhub.tk with subject
"Re: [neovim] Scope variables in move.c (#3297)"
will be delivered as soon as possible.

Your email address is automatically added to the
recipients 'whitelist'. This means for further
emails to danielm@tinyhub.tk, you will not receive a delivery
notification again.

TK MAILIAS
*************************************************
Looking for a way to Protect Your Identity and 
for a great no-more-spam solution for your email
address reply@reply.github.com? Sign up for TK Mailias.

It's free! Please visit http://www.mail.tk/

**Pull request merged**